.github/workflows/kitctl-test.yml

```diff
@@ -0,0 +1,215 @@
+              echo "Controlplane deleted."
+            else
+              echo "No controlplane found  $GUEST_CLUSTER_NAME."
+            fi
+          done 
+          #Adding sleep to ensure that the resources are deleted like EG:Load balancer.
+          sleep 120
+        continue-on-error: true
+        working-directory: substrate
+      - name:  Delete the kit environment
+        if: always()
+        run: |
+          #During the first attempt, "kitctl" was unable to delete some resources sporadically. To ensure that enough time is given for the resources to be deleted, wait time is added around 280 seconds before attempting to delete the kit again.
+          #and for the second iteration with a wait time of 30 seconds to ensure all resources are successfully deleted as we are left with few resources sometimes.
+          #TODO: we should fix `kitctl delete` to make it more robust and not hang forever.
+          cd ${HOME}/work/kubernetes-iteration-toolkit/kubernetes-iteration-toolkit/substrate/cmd/kitctl
+          if [[ "${{ matrix.os }}" == "ubuntu-latest" ]]; then
+            os_type="linux"    
+          elif [[ "${{ matrix.os }}" == "macos-latest" ]]; then
+            os_type="darwin"
+            # installing coreutils for macos to use gtimeout as timeout is not available in macos by default,as kitctl hangs sparodically 
+            /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+            brew install coreutils
+          fi 
+          export KUBECONFIG=${HOME}/.kit/env/kit-github-${os_type}-${{ github.run_id }}-${{github.run_attempt}}/etc/kubernetes/admin.conf
+          # Set the maximum number of attempts
+          MAX_ATTEMPTS=2
+          # Set the delay between retries in seconds
+          RETRY_DELAY=30
+          # Set the maximum duration in seconds
+          MAX_DURATION=280
+          # Set the start time
+          START_TIME=$(date +%s)
+          # Set a flag to indicate whether the command was successfully completed or not
+          COMMAND_COMPLETED=false
+          # Attempt to delete the resource multiple times
+          for i in $(seq 1 $MAX_ATTEMPTS); do
+            # Run the command with a timeout
+            if [[ "${{ matrix.os }}" == "macos-latest"  ]]; then
+              gtimeout $MAX_DURATION ./kitctl delete kit-github-${os_type}-${{ github.run_id }}-${{github.run_attempt}}
+            else
+              timeout $MAX_DURATION ./kitctl delete kit-github-${os_type}-${{ github.run_id }}-${{github.run_attempt}}
+            fi
+            # Check if the command timed out
+            status=$?
+            if [ $status -eq 124 ]; then
+              echo "Command timed out. Retrying in $RETRY_DELAY seconds."
+            elif [ $status -eq 143 ]; then
+              echo "Command terminated by SIGTERM signal."  
+            else
+              echo "Command timed out successfully for $i iteration."
```

Comment:
Earlier comment is not addressed. It will fall into this else clause when exit code is 0. And it will produce confusing log message.